### PR TITLE
Migrate shared classes to kotlin

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #532 

Migrates common classes to Kotlin.

@JSunde PTAL?

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+abstract class AbstractDialogFragment : AppCompatDialogFragment() {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    /**
+     * Uses [ViewModelFactory] to obtain an instance of the view model of the specified class.
+     */
+    protected fun <T : ViewModel?> getViewModel(modelClass: Class<T>): T {
+        return viewModelFactory.get(this, modelClass)
```
Comment (JSunde) [inline gnd/src/main/java/com/google/android/gnd/ui/common/AbstractDialogFragment.kt:39]: nit: could be expressed as a single-line expression function

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractFragment.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import com.google.android.gnd.util.Debug
+import javax.inject.Inject
+
+abstract class AbstractFragment : Fragment() {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    protected fun <T : ViewModel> getViewModel(modelClass: Class<T>): T {
+        return viewModelFactory.get(this, modelClass)
```
Comment (JSunde) [inline gnd/src/main/java/com/google/android/gnd/ui/common/AbstractFragment.kt:36]: ditto

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractFragment.kt [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    private fun <T> restoreChildFragment(savedInstanceState: Bundle, key: String): T {
+        return childFragmentManager.getFragment(savedInstanceState, key) as T
+    }
+
+    protected fun <T> restoreChildFragment(
+        savedInstanceState: Bundle,
+        fragmentClass: Class<T>
+    ): T {
+        return restoreChildFragment<Any>(savedInstanceState, fragmentClass.name) as T
```
Comment (JSunde) [inline gnd/src/main/java/com/google/android/gnd/ui/common/AbstractFragment.kt:126]: ditto

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractView.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            var context = context
+            while (context is ContextWrapper) {
+                if (context is FragmentActivity) return context
+                context = context.baseContext
+            }
+            throw IllegalStateException("View is not contained in FragmentActivity")
+        }
+
+    protected fun <T : ViewModel?> getViewModel(modelClass: Class<T>): T {
+        return viewModelFactory.get(activity, modelClass)
```
Comment (JSunde) [inline gnd/src/main/java/com/google/android/gnd/ui/common/AbstractView.kt:45]: ditto

File: gnd/src/main/java/com/google/android/gnd/ui/common/EphemeralPopups.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        Toast.makeText(context, message, Toast.LENGTH_LONG).show()
+    }
+
+    // TODO: Rename to unknownError?
+    fun showError() {
+        showLong(R.string.unexpected_error)
+    }
+
+    private fun showLong(@StringRes messageId: Int) {
+        Toast.makeText(context, messageId, Toast.LENGTH_LONG).show()
```
Comment (JSunde) [inline gnd/src/main/java/com/google/android/gnd/ui/common/EphemeralPopups.kt:51]: ditto

**Review by JSunde — APPROVED** [review]

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractDialogFragment.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/common/AbstractDialogFragment.kt]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractFragment.kt (hunk 3 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/common/AbstractFragment.kt]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/common/AbstractView.kt (hunk 4 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/common/AbstractView.kt]: Done

**Pull request merged**